### PR TITLE
Add UI Automation Utilities

### PR DESCRIPTION
**Pull request by shahzaibj**
In this PR, I have added a bunch of utility classes that will be used to facilitate writing UI Automation tests.

The following changes are made:

- Add new module called `uiautomationutilities` to common
- Add `UI Automator` as a dependency. This is the framework driving our UI Tests
- Add utility methods for handling prompts during authorization request
- Add utility methods for installing apps from Play Store
- Add methods for interacting with ADB shell
- Add methods for interacting with Broker and perform operations such as device registration

For now these are being used in MSAL UI Tests here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1047

Going forward, we will also be using these for Broker UI Tests.

### CODE REVIEW

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    void uninstall();
+
+    /**
+     * Handle the first run experience for this app on first time launch
+     */
+    void handleFirstRun();
+
+    boolean hasPermission(String permission);
+
+    void grantPermission(String permission);
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java:54]: need javadoc

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    @Override
+    public void performDeviceRegistration(String username, String password) {
+        performDeviceRegistrationHelper(
+                username,
+                password,
+                "com.azure.authenticator:id/manage_device_registration_email_input",
+                "com.azure.authenticator:id/manage_device_registration_register_button"
+        );
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java:69]: needs TODO after this block as there is probably unfinished business here

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/IAppInstaller.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.microsoft.identity.client.ui.automation.installer;
+
+public interface IAppInstaller {
+
+    void installApp(final String appHint);
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/IAppInstaller.java:5]: needs javadoc

File: keyvault/src/main/java/com/microsoft/identity/internal/test/keyvault/ApiClient.java [hunk 4]
```diff
@@ -93,6 +93,9 @@ public ApiClient() {
         authentications.put("oauth", oAuthAuthentication);
         // Prevent the authentications from being modified.
         authentications = Collections.unmodifiableMap(authentications);
+        List<Protocol> protocols = new ArrayList<>();
```
Comment (shoatman) [inline keyvault/src/main/java/com/microsoft/identity/internal/test/keyvault/ApiClient.java:96]: I'd suggest adding a comment here around why this is necessary... mostly so no one removes without testing.

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            assert deviceRegistration != null;
+
+            // click the device registration button
+            deviceRegistration.click();
+
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+                grantPermission(Manifest.permission.GET_ACCOUNTS);
+            }
+        } catch (UiObjectNotFoundException e) {
+            e.printStackTrace();
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java:124]: needs to fail here

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction;
+
+public class MicrosoftPromptHandler extends AbstractPromptHandler {
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java:25]: nit: rename to `AzureActiveDirectoryPromptHandler` or `AadPromptHandler`

File: gradle/versions.gradle [hunk 7]
```diff
@@ -12,6 +12,9 @@ ext {
     androidMavenGradlePluginVersion = "1.4.1"
 
     // Libraries
+    androidxTestRunnerVersion = "1.2.0"
```
Comment (iambmelt) [inline gradle/versions.gradle:15]: Don't forget to make corresponding update in `android-complete` :)

Comment (iambmelt) [reply gradle/versions.gradle:15]: (For all these deps)

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabConfig.java [hunk 8]
```diff
@@ -1,6 +1,9 @@
 package com.microsoft.identity.internal.testutils.labutils;
```
Comment (iambmelt) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabConfig.java:1]: Missing license info?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.app;
+
+public interface IApp {
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java:25]: Interface description?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+     * Remove this app from the device
+     */
+    void uninstall();
+
+    /**
+     * Handle the first run experience for this app on first time launch
+     */
+    void handleFirstRun();
+
+    boolean hasPermission(String permission);
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java:52]: javadoc here too -- are these android permissions? something else? not clear at a glance

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+                .descriptionContains("Shared Device Mode")
+                .className("android.widget.ImageView");
+
+        //confirm that we are in Shared Device Mode inside Authenticator
+        UiObject sharedDeviceConfirmation = mDevice.findObject(sharedDeviceConfirmationSelector);
+        sharedDeviceConfirmation.waitForExists(TIMEOUT);
+        Assert.assertTrue(sharedDeviceConfirmation.exists());
+    }
+
+    private void performDeviceRegistrationHelper(final String username,
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java:94]: Just call out: love the use of comments here. Makes this super readable

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    public final static String COMPANY_PORTAL_APK = "CompanyPortal.apk";
+
+    public BrokerCompanyPortal() {
+        super(COMPANY_PORTAL_APP_PACKAGE_NAME, COMPANY_PORTAL_APP_NAME);
+        localApkFileName = COMPANY_PORTAL_APK;
+    }
+
+    @Override
+    public void performDeviceRegistration(String username, String password) {
+        //TODO implement device registration for CP
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:54]: Do we need to file work items/throw unimplemented exceptions?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Override
+    public void performSharedDeviceRegistration(String username, String password) {
+        //TODO implement shared device registration for CP
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:59]: Do we need to file work items/throw unimplemented exceptions?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Override
+    public void handleFirstRun() {
+        //TODO handle first run for CP
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:64]: Do we need to file work items/throw unimplemented exceptions?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/browser/BrowserChrome.java [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.browser;
+
+import com.microsoft.identity.client.ui.automation.app.App;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+public class BrowserChrome extends App {
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/browser/BrowserChrome.java:28]: Would be good to javadoc `App.java`; if the superclass has clear docs, the subclasses get easier to grok

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/LocalApkInstaller.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
+package com.microsoft.identity.client.ui.automation.installer;
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/LocalApkInstaller.java:1]: License info

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java [hunk 17]
```diff
@@ -0,0 +1,124 @@
+package com.microsoft.identity.client.ui.automation.installer;
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java:1]: License info

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java (hunk 6 above)
Comment (iambmelt) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java:25]: +1 on `AadPromptHandler`

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.microsoft.identity.client.ui.automation.broker.ITestBroker;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+
+@Builder
+@Getter
+/**
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:33]: Javadoc goes before the `@Annotations`

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    private boolean speedBumpExpected;
+
+    // whether we are expecting at least one account in broker
+    private boolean expectingNonZeroAccountsInBroker;
+
+    // whether we are expecting at least one account in the browser/webview cookie
+    private boolean expectingNonZeroAccountsInCookie;
+
+    @Builder.Default
+    // The way in which we want to respond to consent page for this request
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:65]: Same here, comments between the annotation and its target... should be flipped around

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+    private UiResponse consentPageResponse = UiResponse.ACCEPT;
+
+    @Builder.Default
+    // The way in which we want to respond to speed bump page for this request
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:69]: Same here, comments between the annotation and its target... should be flipped around

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    /**
+     * Remove the supplied package name from the device
+     *
+     * @param packageName the package name to remove
+     */
+    public static void removePackage(final String packageName) {
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java:27]: Lots of places without any nullability annots -- these would be good to have. Will just call out once here

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+
+import androidx.test.core.app.ApplicationProvider;
+
+public class CommonUtils {
+
+    public final static long TIMEOUT = 1000 * 30;
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java:33]: Could consider doing `TimeUnit.SECONDS.toMillis(30)` but is minor

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Launch (open) the supplied package on the device
+     *
+     * @param packageName the package name to launch
+     */
+    public static void launchApp(final String packageName) {
+        final Context context = ApplicationProvider.getApplicationContext();
+        Intent intent = context.getPackageManager().getLaunchIntentForPackage(packageName);  //sets the intent to start your app
```
Comment (iambmelt) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java:42]: nit: `final`

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java (hunk 6 above)
Comment (shoatman) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java:25]: Is it AAD v1 or v2?  If v2 i'd mirror what we do relative to the strategy for v2... which i think is MIcrosoftSTS or something like that.

Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java:25]: It is v2, I would personally want to stick to `AadPromptHandler` as we will need different prompt handler for different policies and identity providers (once we get to writing those tests). Will probably need a different one for ADFS as well.

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java (hunk 1 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java (hunk 2 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java:71]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/IAppInstaller.java (hunk 3 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/IAppInstaller.java:35]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+
+import androidx.test.uiautomator.UiDevice;
+
+import org.junit.Assert;
+
+import java.io.IOException;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+
+public class AdbShellUtils {
```
[the remark this answers is not in the document]
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java:16]: Addressed

File: keyvault/src/main/java/com/microsoft/identity/internal/test/keyvault/ApiClient.java (hunk 4 above)
Comment (shahzaibj) [reply keyvault/src/main/java/com/microsoft/identity/internal/test/keyvault/ApiClient.java:101]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java (hunk 5 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerAuthenticator.java]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java (hunk 6 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/MicrosoftPromptHandler.java]: Addressed by renaming to `AadPromptHandler`

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabConfig.java (hunk 8 above)
Comment (shahzaibj) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabConfig.java:23]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java (hunk 9 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java:28]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java (hunk 10 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java:61]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java (hunk 12 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:56]: Addressed by throwing exceptions. I will also create work items later

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java (hunk 13 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:63]: Addressed by throwing exceptions. I will also create work items later

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java (hunk 14 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:69]: Addressed by throwing exceptions. I will also create work items later

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/browser/BrowserChrome.java (hunk 15 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/browser/BrowserChrome.java:28]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/LocalApkInstaller.java (hunk 16 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/LocalApkInstaller.java:23]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java (hunk 17 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java:23]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java (hunk 18 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:31]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java (hunk 19 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:64]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java (hunk 20 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/PromptHandlerParameters.java:68]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java (hunk 21 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/AdbShellUtils.java]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java (hunk 22 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java (hunk 23 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/CommonUtils.java]: Addressed

File: gradle/versions.gradle (hunk 7 above)
Comment (shahzaibj) [reply gradle/versions.gradle:15]: Opened up a PR here: https://github.com/AzureAD/android-complete/pull/41

**Comment by shahzaibj** [comment]
@iambmelt Thanks a lot for the review! All of your comments have been addressed

**Pull request merged**